### ISSUE TITLE
Intermittent Pod Restarts with PyTorch Plugin in Volcano Jobs

### ISSUE THREAD

**Issue by daniel-hutao** [volcano-sh/volcano#3289]
**What happened**:

I've encountered an issue while following the instructions in the PyTorch plugin user guide (https://github.com/volcano-sh/volcano/blob/master/docs/user-guide/how_to_use_pytorch_plugin.md.

After applying the job YAML as per the documentation, the pods were repeatedly restarting. Interestingly, they eventually succeeded after numerous restarts (dozens of times). This behavior seems like a potential bug.

<img width="1307" alt="image" src="https://github.com/volcano-sh/volcano/assets/18692628/b273248f-a362-4ed1-9215-05cceb7b6948">

<img width="1504" alt="image" src="https://github.com/volcano-sh/volcano/assets/18692628/a89f462b-3f9a-47f7-95f3-3e08011f644b">

However, when I added the `svc` plugin to the job configuration, the pods started successfully on the first attempt. This leads me to believe that there might be an issue with the PyTorch plugin.

<img width="1793" alt="image" src="https://github.com/volcano-sh/volcano/assets/18692628/814640bc-cf12-4f6c-bf8c-8ec63d5809d9">

**How to reproduce it (as minimally and precisely as possible)**:

Apply the yaml given by the document above.

**Environment**:
- Volcano Version: latest
- Kubernetes version (use `kubectl version`): v1.24.10

**Comment by daniel-hutao** [comment]
I retried many times, and finally, I found that with a worker count of 10 and the `svc` configuration, it still fails. So, the issue is not determined by whether the `svc` plugin exists or not; it's a problem with the code logic inside the container, and the code should have a retry mechanism instead of restart always.

**Comment by Monokaix** [comment]
> I retried many times, and finally, I found that with a worker count of 10 and the `svc` configuration, it still fails. So, the issue is not determined by whether the `svc` plugin exists or not; it's a problem with the code logic inside the container, and the code should have a retry mechanism instead of restart always.

Yeah, seems ps and woker should guarantee start up order itself, or kubelet will restart them until success.
